sdk/core/core-lro/test/utils/router/paramRoutes.ts

```diff
@@ -0,0 +1,384 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { createHttpHeaders, PipelineRequest, PipelineResponse } from "@azure/core-rest-pipeline";
+
+import { buildResponse, getPascalCase, parseUri } from "./utils";
+
+function putBody(request: PipelineRequest): PipelineResponse | undefined {
+  function isValidRequest(
+    initialCode: number,
+    initialState: string,
+    finalState: string,
+    finalCode: number
+  ): boolean {
+    return (
+      (initialCode === 201 &&
+        initialState === "Creating" &&
+        finalState === "Succeeded" &&
+        finalCode === 200) ||
+      (initialCode === 200 &&
+        initialState === "Updating" &&
+        finalState === "Succeeded" &&
+        finalCode === 200) ||
+      (initialCode === 201 &&
+        initialState === "Created" &&
+        finalState === "Failed" &&
+        finalCode === 200) ||
+      (initialCode === 200 &&
+        initialState === "Accepted" &&
+        finalState === "Canceled" &&
+        finalCode === 200)
+    );
+  }
+  const path = request.url;
+  const pieces = path.substr(1).split("/");
+  if (pieces.length === 5) {
+    try {
+      const initialCode: number = JSON.parse(pieces[1]);
```

Comment:
Can we wrap all of the JSON.parse calls into try catch blocks ? (or to have a parser function that would deal with the try-catch of all of the JSON.parses). Ideally we would have known failure cases with known error messages in the cases the JSONs are misshapen.